tests/gpu_provers.rs

```diff
@@ -76,7 +76,7 @@ pub fn test_parallel_prover() {
     let pvk2 = prepare_verifying_key(&params2.vk);
 
     let higher_thread = thread::spawn(move || {
```

Comment:
Currently the verification is manually, by looking at the logs and looking for a warning message like 

    [2022-12-01T09:33:52Z WARN  bellperson::gpu::locks] GPU acquired by a high priority process! Freeing up Multiexp kernels...

Your idea sounds good, it might be a way to properly test this reliably (which I'd really love to see). Please note that in some cases (IIRC), the low priority tests are then run on the CPU. In theory the CPU code path could be faster (you never know on CI), but I'd hope it wouldn't. Do you want to give it a try?

Reply:
Yeah I would like to take a look at this.  I'll open a new issue rather than hold up this patch.

Reply:
That would be great, thatnks!